File: skiCentar/skiCentar.Model/SkiAccident.cs
using System;

namespace skiCentar.Model
{
    public class SkiAccident
    {
        public int Id { get; set; }

        public int? UserId { get; set; }

        public int? TrailId { get; set; }

        public int? PeopleInvolved { get; set; }

        public bool? IsReporterInjured { get; set; }

        public bool? IsActive { get; set; }

        public decimal? LocationX { get; set; }

        public decimal? LocationY { get; set; }

        public DateTime? Timestamp { get; set; }

        public string Description { get; set; } = null!;

        public virtual Trail? Trail { get; set; }

        public virtual User? User { get; set; }
    }
}

File: skiCentar/UI/skicentar_mobile/lib/models/ski_accident.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:skicentar_mobile/models/trail.dart';
import 'package:skicentar_mobile/models/user.dart';

part 'ski_accident.g.dart';

@JsonSerializable()
class SkiAccident {
  int? id;
  String? description;
  bool? isActive;
  bool? isReporterInjured;
  int? userId;
  int? trailId;
  int? peopleInvolved;
  double? locationX;
  double? locationY;
  DateTime? timestamp;

  Trail? trail;
  User? user;

  SkiAccident({this.id, this.locationX, this.locationY});

  factory SkiAccident.fromJson(Map<String, dynamic> json) =>
      _$SkiAccidentFromJson(json);

  Map<String, dynamic> toJson() => _$SkiAccidentToJson(this);
}
